dpcpp/CMakeLists.txt

```diff
@@ -51,11 +51,16 @@ target_sources(ginkgo_dpcpp
 
 ginkgo_compile_features(ginkgo_dpcpp)
 
-set(GINKGO_DPCPP_FLAGS ${GINKGO_COMPILER_FLAGS} -fsycl -fsycl-device-lib=libc)
+set(GINKGO_DPCPP_FLAGS ${GINKGO_COMPILER_FLAGS} -fsycl)
 set(GINKGO_DPCPP_FLAGS ${GINKGO_DPCPP_FLAGS} PARENT_SCOPE)
 target_compile_options(ginkgo_dpcpp PRIVATE "${GINKGO_DPCPP_FLAGS}")
 target_compile_features(ginkgo_dpcpp PRIVATE cxx_std_17)
+target_link_options(ginkgo_dpcpp PRIVATE -fsycl-device-lib=all)
+target_link_options(ginkgo_dpcpp PRIVATE -fsycl-device-code-split=per_kernel)
 target_link_libraries(ginkgo_dpcpp PUBLIC ginkgo_device)
+if (GINKGO_DPCPP_SINGLE_MODE)
```

Comment:
Because `-fsycl-device-code-split=per_kernel`, should we separate the SINGLE_MODE to build/test usage?
To test full build on float-only device

Reply:
Sorry, I do not understand this one.

Reply:
we add the link flag into ginkgo library, which allows us to use single kernel (test) on single-only device even if compiling ginkgo with double kernel